toolkit/tools/scheduler/schedulerutils/buildworker.go

```diff
@@ -186,6 +189,35 @@ func getBuildDependencies(node *pkggraph.PkgNode, pkgGraph *pkggraph.PkgGraph, g
 	return
 }
 
+func parseCheckSection(logFile string) (checkErr error) {
+	file, logErr := os.Open(logFile)
+	// If we can't open the log file, that's a build error.
+	if logErr != nil {
+		logger.Log.Errorf("Failed to open log file '%s' while checking package test results. Error: %v", logFile, logErr)
+		checkErr = logErr
+		return
+	}
+	defer file.Close()
+	for scanner := bufio.NewScanner(file); scanner.Scan(); {
+		currLine := scanner.Text()
+		// Anything besides 0 is a failed test
+		if strings.Contains(currLine, "CHECK DONE") && !strings.Contains(currLine, "EXIT STATUS 0") {
+			failedLogFile := fmt.Sprintf("%s-FAILED_TEST-%d.log", logFile[:strings.Index(logFile, ".log")], time.Now().UnixMilli())
+			checkErr = os.Rename(logFile, failedLogFile)
+			if checkErr != nil {
+				logger.Log.Errorf("Log file rename failed. Error: %v", checkErr)
+				return
+			}
+			checkErr = fmt.Errorf("Package test failed. Test status line: %s", currLine)
+			return
+		}
+		if strings.Contains(currLine, "CHECK DONE") && strings.Contains(currLine, "EXIT STATUS 0") {
+			return
+		}
```

Comment:
Good idea, but let's combine it with the check above, since both begin with `strings.Contains(currLine, "CHECK DONE")`.